### PR TITLE
Point Discover canonical at /discover and qualify signup SERP fees

### PR DESCRIPTION
**Pull request by gumclaw**
> Draft status: hosted stills of `/discover` + `/login` + `/signup` landed at `ebffb7dc71`. Remaining gate is a human copy pass on the signup fee sentence. Not auto-mergeable until that lands.

## What
`/discover` was emitting a canonical that pointed at `/` on the discover host (the marketing homepage), so Google treated Discover as a duplicate of home. Login and signup had no meta description.

This branch:
- Points the Discover canonical at `/discover`
- Adds login/signup SERP snippets
- Qualifies the signup fee sentence as **direct 10% + 50¢** vs **Discover 30%**, matching `/pricing` and the home meta (unqualified "10% + 50¢ per sale" overclaims the marketplace rate)
- Pins login/signup canonicals to `UrlService.domain_with_protocol` so `app.gumroad.com` does not advertise itself as canonical

## Why
Discover was de-indexing itself. Signup SERP copy stated the direct fee as if it were every sale. Both `gumroad.com` and `app.gumroad.com` serve `/login` and `/signup`; a request-host canonical would split those snippets.

## Before / after
- Before: Discover canonical → `/` on the discover host. Login/signup had no description. After the first commits, login/signup canonicals followed the request host (`app.gumroad.com/login` vs `gumroad.com/login`).
- After: Discover canonical → `/discover`. Signup description names both fee channels. Login/signup canonicals are always `{PROTOCOL}://{DOMAIN}/login` and `/signup`.

## Checklist
- [x] Scope — Discover canonical/title/breadcrumb + login/signup SERP meta. No money-path change.
- [x] Design — pin auth canonicals to `UrlService.domain_with_protocol` rather than `login_url`/`signup_url` (those inherit the request host; passing `host: DOMAIN` also drops the test port when the request has none).
- [x] Build — `7723b80d25` on `gumclaw/seo-discover-canonical-and-auth-meta`
- [x] QA — hosted stills at `ebffb7dc71` (see below). Copy pass on the fee sentence still owed.
- [ ] Shipped
- [ ] Market — n/a (SERP metadata, no launch)
- [ ] Sell — n/a

## Test Results
At `7723b80d25`, `DISABLE_SPRING=1 bundle exec rspec spec/controllers/logins_controller_spec.rb:26 spec/controllers/signup_controller_spec.rb:99 spec/controllers/signup_controller_spec.rb:110` — 3 examples, 0 failures.

Mutation: replace the pin with `login_url`/`signup_url`, same two host-pin examples — both red (`http://127.0.0.1/login` vs `http://app.test.gumroad.com:31337/login`, same for signup). Restored before commit.

## QA steps
1. View-source `/discover` on the preview — canonical href is `/discover`, not `/`.
2. View-source `/signup` — meta description names both the direct 10% + 50¢ rate and the 30% Discover rate; canonical host is `gumroad.com`, not `app.gumroad.com`.
3. View-source `/login` — description is present; same host pin.
4. View-source `/login` on `app.gumroad.com` if the preview serves it — canonical still names the public host.

Hosted stills at `https://gumclaw-seo-discover-canonical-a.apps.staging.gumroad.org` (x-revision `7723b80d25`):

- Discover canonical is `/discover`.
- discover
- login meta
- signup meta

DOM-read at that host: `<link rel="canonical" href="…/discover">`; login description `Log in to your Gumroad account…`; signup description names `Direct sales: 10% + 50¢. Discover: 30%.`

---

AI disclosure: grok-4.6. Prompt: handle the Greptile P1 on gumroad#7292 (canonical host varies by request).

Premerge review: clean @ ebffb7dc7149dcf990665d2d43a2b34019bf8c53

### CODE REVIEW

**Comment by gumclaw** [comment]
Draft on purpose: stills of `/discover` canonical + signup/login SERP meta are still owed, and the fee sentence needs a human copy pass. Opening this now because the branch sat 10h with no PR after yesterday's `gh pr create` 503.

File: app/controllers/logins_controller.rb [hunk 1]
```diff
@@ -20,7 +20,9 @@ class LoginsController < Devise::SessionsController
   def new
     return redirect_to login_path(next: request.referrer) if params[:next].blank? && request_referrer_is_a_valid_after_login_path?
 
-    set_meta_tag(title: "Log In")
+    set_meta_tag(title: "Log in to Gumroad")
+    set_meta_tag(name: "description", content: "Log in to your Gumroad account to access your dashboard, library, and sales.")
+    set_meta_tag(tag_name: "link", rel: "canonical", href: login_url, head_key: "canonical")
```
[the remark this answers is not in the document]
Comment (gumclaw) [reply app/controllers/logins_controller.rb]: Fixed in 7723b80d25. Login and signup canonicals now use UrlService.domain_with_protocol so app.gumroad.com no longer advertises itself as the preferred URL.

**Comment by gumclaw** [comment]
Hosted stills landed at `ebffb7dc71`.

`https://gumclaw-seo-discover-canonical-a.apps.staging.gumroad.org` at x-revision `7723b80d25`:
- `/discover` canonical is `/discover` (not `/`)
- `/login` and `/signup` both have meta descriptions; signup names direct 10% + 50¢ vs Discover 30%

Still draft for the human copy pass on that fee sentence.

**Comment by gumclaw** [comment]
Readied. Hosted stills and premerge are on the head (`ebffb7dc71`). Relevant CI is green.

@gianfrancopiana the remaining call is the signup SERP fee sentence (`Direct sales: 10% + 50¢. Discover: 30%.`) — it already matches `/pricing` and the home meta. Need a copy/merge pass, not more code from me.
